src/public_parameters/registry.rs

```diff
@@ -0,0 +1,91 @@
+use std::{
+    collections::{hash_map::Entry, HashMap},
+    sync::{Arc, Mutex},
+};
+
+use once_cell::sync::Lazy;
+use pasta_curves::pallas;
+use serde::{de::DeserializeOwned, Serialize};
+use tap::TapFallible;
+
+use crate::public_parameters::Error;
+use crate::{coprocessor::Coprocessor, eval::lang::Lang, proof::nova::PublicParams};
+
+use super::file_map::FileIndex;
+
+type S1 = pallas::Scalar;
+type AnyMap = anymap::Map<dyn anymap::any::Any + Send + Sync>;
+type PublicParamMemCache<C> = HashMap<usize, Arc<PublicParams<'static, C>>>;
+
+/// This is a global registry for Coproc-specific parameters.
+/// It is used to cache parameters for each Coproc, so that they are not
```

Comment:
I think parameters vary by `Lang`, not by `Coproc`, right?

Reply:
#386